retrieval/discovery/consul.go

```diff
@@ -191,7 +184,7 @@ func (cd *ConsulDiscovery) watchServices(update chan<- *consulService) {
 		})
 		if err != nil {
 			log.Errorf("Error refreshing service list: %s", err)
-			<-time.After(consulRetryInterval)
+			time.Sleep(consulRetryInterval)
```

Comment:
Not for this PR but can probably reuse the `until` in kubernetes discovery to shut down target providers nicely (https://github.com/prometheus/prometheus/blob/master/retrieval/discovery/kubernetes/discovery.go#L538-L555). Borrowed from kubernetes codebase btw ;) Worth making it a common util?